### ISSUE TITLE
feat: video support

### ISSUE THREAD

**Issue by niceEli** [kaplayjs/kaplay#157]
idk if useful but would be good if you could have a node that plays a video

```ts
k.loadSprite("testVid", "./mp3s/testVid.mov")
```

```ts
k.add([
  k.sprite("testVid")
]);
```

and also the controls like with audio component would be in video component (like pause, play, loop, detune, start position)

**Comment by amyspark-ng** [comment]
i think this is awesome, and should be added

**Comment by mflerackers** [comment]
I think playing a video on a texture would be most powerful? Since you can still use shaders and vertex deformation?

**Comment by SpcFORK** [comment]
> I think playing a video on a texture would be most powerful? Since you can still use shaders and vertex deformation?

Yes, would be nice

**Comment by totomtornado** [comment]
yes... same server... local server...

in the kaplay.js... it hits here:  line 3539
function Ii(t, e) {
  let r = document.createElement("video"), n = false, o = false, s = false, a = new xe(u.gfx.ggl, e.width, e.height);

but the element is obviously empty...

**Comment by mflerackers** [comment]
No, the element is r. What is undefined is e, the opt, you probably use video(url) instead of video(url, { width:320, height: 200 });

**Comment by totomtornado** [comment]
> No, the element is r. What is undefined is e, the opt, you probably use video(url) instead of video(url, { width:320, height: 200 });

specifying width and height worked... am I looking at the wrong documentation?  I looked at the link above... might have just not been updated but if there is a better source for documentation then please let me know... thanks for the reply!!!

**Comment by totomtornado** [comment]
I would definitely be interested in exposing settings/attributes like loop, autoplay, muted, etc... because it is a bit difficult to start the video on page load unless you set some of these features.

**Comment by lajbel** [comment]
Docs are not updated. I will setup a documentation for master, because we don't continuous releae
